File: lingting-aws/src/main/java/live/lingting/framework/aws/s3/interfaces/AwsS3Delegation.java
package live.lingting.framework.aws.s3.interfaces;

import live.lingting.framework.aws.AwsS3Client;

/**
 * @author lingting 2024-09-19 22:06
 */
public interface AwsS3Delegation<C extends AwsS3Client> {

	C delegation();

}

File: lingting-aws/src/main/kotlin/live/lingting/framework/aws/s3/interfaces/AwsS3Delegation.kt
package live.lingting.framework.aws.s3.interfaces

import live.lingting.framework.aws.AwsS3Client

/**
 * @author lingting 2024-09-19 22:06
 */
interface AwsS3Delegation<C : AwsS3Client?> {
    fun delegation(): C
}
